File: Python_Collections_benchmark/Benchmark.py
from IBenchmark import IBenchmark
from BenchmarkResults import BechmarkResults
from time import time


class Benchmark:
    def __init__(self, ibenchmark: IBenchmark):
        self.ibenchmark = ibenchmark
        self.results = BechmarkResults()

    def perform_all_tests(self):
        self._perform_add()
        self._perform_find()
        self._perform_remove()

        return self.results

    def _perform_add(self):
        start = time()
        self.ibenchmark.add_test()
        end = time()
        self.results.add_time = (end - start) * 1000.0

    def _perform_find(self):
        start = time()
        self.ibenchmark.find_test()
        end = time()
        self.results.find_time = (end - start) * 1000.0

    def _perform_remove(self):
        start = time()
        self.ibenchmark.remove_test()
        end = time()
        self.results.remove_time = (end - start)* 1000.0

File: CS_Collections_benchmark/Benchmark.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Diagnostics;

namespace CS_Collections_benchmark
{
    class Benchmark
    {
        IBenchmark benchmark;
        BenchmarkResults results;
        Stopwatch sw = new Stopwatch();

        public Benchmark(IBenchmark benchmark)
        {
            this.benchmark = benchmark;
            results = new BenchmarkResults();
        }

        public BenchmarkResults PerformAllTests()
        {
            PerformAdd();
            PerformFind();
            PerformRemove();

            return results;
        }

        private void PerformAdd()
        {
            sw.Restart();
            benchmark.AddTest();
            sw.Stop();
            results.AddTime = sw.Elapsed.TotalMilliseconds;
            Console.WriteLine(results.AddTime);
        }
        private void PerformFind()
        {
            sw.Restart();
            benchmark.FindTest();
            sw.Stop();
            results.FindTime = sw.Elapsed.TotalMilliseconds;
            Console.WriteLine(results.FindTime);
        }

        private void PerformRemove()
        {
            sw.Restart();
            benchmark.RemoveTest();
            sw.Stop();
            results.RemoveTime = sw.Elapsed.TotalMilliseconds;
            Console.WriteLine(results.RemoveTime);
        }
    }
}
